js/router.js

```diff
@@ -3,13 +3,17 @@
 'use strict';
 
 const router = {
-  buffer: [],
+  buffer: (history.state || {}).buffer || [],
   watchers: [],
 
   getSearch(key) {
     return new URLSearchParams(location.search).get(key);
   },
 
+  push(url, state = history.state) {
+    history.pushState(Object.assign({buffer: router.buffer}, state), null, url);
```

Comment:
Should we put `state` object before `{buffer}`?

Reply:
Why? It would add buffer into state.

Reply:
Like `Object.assign({}, state, {buffer: router.buffer})` so we always use  `router.buffer` instead of `state.buffer`.

Reply:
I see but we don't have buffer in state. If we ever do we should not overwrite it either so we will use {buffer: router.buffer, state: history: state}

Reply:
Ah, damn, we will have buffer in state after this line.